lib/afinter.c

```diff
@@ -496,3 +501,77 @@ afinter_global_deinit(void)
     }
   current_internal_source = NULL;
 }
+
+/****************************************************************************
+ * Live collection of internal log messages.
+ ****************************************************************************/
+
+guint8
+afinter_start_live_collection(void)
+{
+  AFInterSource *self;
+
+  if (is_live_collection)
+    return AF_LIVE_COLLECTION_RUNNING;
+
+  if (current_internal_source != NULL)
+    return AF_INTERNAL_SRC_PRESENT;
+
+  self = g_new0(AFInterSource, 1);
+
+  g_static_mutex_lock(&internal_msg_lock);
+  current_internal_source = self;
+  g_static_mutex_unlock(&internal_msg_lock);
+
+  is_live_collection = TRUE;
+
+  return AF_LIVE_COLLECTION_STARTED;
+}
+
+void
+afinter_reset_live_collection(void)
+{
+  afinter_stop_live_collection();
+
+  if (internal_msg_queue)
+    _release_internal_msg_queue();
+}
+
+void
+afinter_stop_live_collection(void)
+{
+  if (is_live_collection)
+    {
+      if (current_internal_source != NULL)
+        {
+          AFInterSource *self = current_internal_source;
+
+          g_static_mutex_lock(&internal_msg_lock);
+          current_internal_source = NULL;
+          g_static_mutex_unlock(&internal_msg_lock);
+
+          is_live_collection = FALSE;
+          g_free(self);
+        }
+    }
+}
+
+void
+afinter_get_collected_messages(GString *result)
+{
+  LogMessage *msg;
+  const gchar *msg_text;
+
+  g_static_mutex_lock(&internal_msg_lock);
+
+  while (!g_queue_is_empty(internal_msg_queue))
+    {
+      msg = g_queue_pop_head(internal_msg_queue);
+      msg_text = log_msg_get_value(msg, LM_V_MESSAGE, NULL);
```

Comment:
msg_text will only hold the MESSAGE part of the log message.
We would need a properly formatted log message here, with timestamps, program field, and all.
You can check how the internal logs look when they are written to STDERR when using -Fedv. That should be your model.

Reply:
@jszigetvari : Yes, I have checked debug messages with timestamps, program field etc. 

I think ```log_writer_format_log``` function formats the internal logs. 

Is there any function similar to this, which I can use to format the internal log messages?